scenarios/socket.benchmarks.yml

```
@@ -0,0 +1,134 @@
+# Example:
+# crank --config https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/socket.benchmarks.yml --scenario read-write --profile aspnet-perf-lin
+
+imports:
+  - https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/aspnet.profiles.standard.yml
+
+variables:
+  # common parameters
+  serverPort: 5000
+  sendBufferSize: 32768 # bytes
+  receiveBufferSize: 32768 # bytes
+
+  # client-specific parameters
+  connections: 1 #number of parallel connections
+  warmup: 15 #seconds
+  duration: 15 #seconds
+
+scenarios:
+  read-write:
+    server:
+      job: socketServer
+      agent: secondary
+      variables:
+        scenario: ReadWrite
+    client:
+      job: socketClient
+      agent: main
+      variables:
+        scenario: ReadWrite
+        host: "{{secondaryAddress}}"
+
+  client-read:
+    server:
+      job: socketServer
+      agent: secondary
+      variables:
+        sendBufferSize: 0 # server's receive buffer size -- send and receive sizes are swapped for server side
+        scenario: ReadWrite
+      onConfigure:
+        - '
+          if (String(job.Variables.sendBufferSize) != "0")
+          {
+            throw "Incorrect sendBufferSize for client-read scenario. Expected: 0, got: " + job.Variables.sendBufferSize;
+          }
+          '
+    client:
+      job: socketClient
+      agent: main
+      variables:
+        scenario: ReadWrite
+        host: "{{secondaryAddress}}"
+        sendBufferSize: 0
+      onConfigure:
+        - '
+          if (String(job.Variables.sendBufferSize) != "0")
+          {
+            throw "Incorrect sendBufferSize for client-read scenario. Expected: 0, got: " + job.Variables.sendBufferSize;
+          }
+          '
+
+  client-write:
+    server:
+      job: socketServer
+      agent: secondary
+      variables:
+        receiveBufferSize: 0 # server's write buffer size -- send and receive sizes are swapped for server side
+        scenario: ReadWrite
+      onConfigure:
+        - '
+          if (String(job.Variables.receiveBufferSize) != "0")
+          {
+            throw "Incorrect receiveBufferSize for client-write scenario. Expected: 0, got: " + job.Variables.receiveBufferSize;
+          }
+          '
+    client:
+      job: socketClient
+      agent: main
+      variables:
+        scenario: ReadWrite
+        host: "{{secondaryAddress}}"
+        receiveBufferSize: 0
+      onConfigure:
+        - '
+          if (String(job.Variables.receiveBufferSize) != "0")
+          {
+            throw "Incorrect receiveBufferSize for client-write scenario. Expected: 0, got: " + job.Variables.receiveBufferSize;
+          }
+          '
+
+  connection-establishment:
```

It is generally better to have shorter names, if possible, bc it is easier to use them in command line :)
```suggestion
  connect:
```